iohub/reader.py

```diff
@@ -111,197 +112,70 @@ def _get_sub_dirs(f: str):
-        """
-
-        logging.basicConfig(
-            level=log_level,
-            format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",  # noqa
-        )
-        self.log = logging.getLogger(__name__)
+    Parameters
+    ----------
+    path : str
+        File path, directory path to ome-tiff series, or Zarr root path
+    data_type :
+    Literal["singlepagetiff", "ometiff", "ndtiff", "omezarr"], optional
+        Dataset format, by default None
+    extract_data : bool, optional
+        True if ome_series should be extracted immediately, by default False
+    log_level : int, optional
+        One of 0, 10, 20, 30, 40, 50 for
+        NOTSET, DEBUG, INFO, WARNING, ERROR, CRITICAL respectively,
+        by default logging.WARNING
 
-        # try to guess data type
-        if data_type is None:
-            if ngff_version := _check_zarr_data_type(src):
-                data_type = "zarr"
-            elif _check_pycromanager(src):
-                data_type = "pycromanager"
-            elif _check_multipage_tiff(src):
-                data_type = "ometiff"
-            elif _check_single_page_tiff(src):
-                data_type = "singlepagetiff"
-            else:
-                raise FileNotFoundError(
-                    f"No compatible data found under {src}, "
-                    "please specify the top level micromanager directory."
-                )
-        self.data_type = data_type
+    Returns
+    -------
+    Reader
+        A child instance of ReaderBase
+    """
 
-        # identify data structure type
-        if data_type == "ometiff":
-            self.reader = MicromanagerOmeTiffReader(src, extract_data)
-        elif data_type == "singlepagetiff":
-            self.reader = MicromanagerSequenceReader(src, extract_data)
-        elif data_type == "zarr":
-            self.reader = ZarrReader(src, version=ngff_version)
-        elif data_type == "pycromanager":
-            self.reader = PycromanagerReader(src)
-        elif data_type == "upti":
-            self.reader = UPTIReader(src, extract_data)
+    logging.basicConfig(
+        level=log_level,
+        format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",  # noqa
+    )
+    logging.getLogger(__name__)
+
+    # try to guess data type
+    if data_type is None:
+        if ngff_version := _check_zarr_data_type(path):
+            data_type = "omezarr"
+        elif _check_ndtiff(path):
+            data_type = "ndtiff"
+        elif _check_multipage_tiff(path):
+            data_type = "ometiff"
+        elif _check_single_page_tiff(path):
+            data_type = "singlepagetiff"
         else:
-            raise NotImplementedError(
-                f"Reader of type {data_type} is not implemented"
+            raise FileNotFoundError(
+                f"No compatible data found under {path}, "
+                "please specify the top level micromanager directory."
```

Comment:
Can we improve this message?
https://github.com/mehta-lab/waveorder/issues/93

I think removing the second line will remove much of the confusion.

Reply:
The error message can certainly be improved. I think your suggestions in https://github.com/mehta-lab/waveorder/issues/93 are great!